index.js

```diff
@@ -0,0 +1,106 @@
+const { default: axios } = require('axios');
+const { Handler } = require('htmlmetaparser');
+const { Parser } = require('htmlparser2');
+
+/**
+ * This is a recursive function that returns an array of dataset site URLs.
+ * If the URL supplied is a data catalog collection, it takes all the part collections in hasPart and crawls them.
+ * If the URL supplied is a data catalog, it takes the dataset array and flattens them. 
+ * 
+ * @param {string} dataCatalogUrl
+ * @returns {Promise<string[]>}
+ */
+async function recursivelyGetCatalogsToGetDatasetUrls(dataCatalogUrl) {
```

Comment:
I'm wondering whether this might be clearer, in that it doesn't require a deep understanding of how the data catalogs are parsed to understand what the function does?

```suggestion
async function getAllDatasetSiteUrls(dataCatalogUrl) {
```

Reply:
Sure